### PR TITLE
📖 Fix experiment-related sample code in documentation

### PR DESCRIPTION
**Pull request by caroqliu**
`building-an-amp-extension.md` provides experiment-related code snippets that cause Travis checks to fail.

- Replace forbidden `this.getWin()` with `this.win` and incorrect `user.warn()` with `userAssert()`
- Update imports in sample code
- Replaces `EXPERIMENT` in calls to `isExperimentOn()` with its string literal equivalent (a required change due to check failures)

### CODE REVIEW

File: contributing/building-an-amp-extension.md [hunk 1]
```diff
@@ -679,20 +680,20 @@ Class AmpMyElement extends AMP.BaseElement {
 
   /** @override */
   buildCallback() {
-    if (!isExperimentOn(this.getWin(), EXPERIMENT)) {
-      user.warn('Experiment %s is not turned on.', EXPERIMENT);
-      return;
-    }
+    if(!isExperimentOn(this.win, 'amp-my-element')) {
```
Comment (gmajoulet) [inline contributing/building-an-amp-extension.md]: Nit: here and below, please add a space after `if`. Also, is there any reason why you no longer want to use the constant `EXPERIMENT`?

File: contributing/building-an-amp-extension.md [hunk 2]
```diff
@@ -679,20 +680,20 @@ Class AmpMyElement extends AMP.BaseElement {
...
+      user().warn(TAG, `Experiment ${EXPERIMENT} is not turned on.`);
+      return();
```
Comment (gmajoulet) [inline contributing/building-an-amp-extension.md]: Nit: here and below, please remove the `()`

File: contributing/building-an-amp-extension.md [hunk 3]
```diff
@@ -679,20 +680,20 @@ Class AmpMyElement extends AMP.BaseElement {
...
+    }
```
Comment (gmajoulet) [inline contributing/building-an-amp-extension.md]: Nit: here and below, extra space

File: contributing/building-an-amp-extension.md (hunk 1 above)
Comment (caroqliu) [reply contributing/building-an-amp-extension.md]: Yes -- passing the constant fails lint with error `isExperimentOn must be passed an explicit string`

Comment (gmajoulet) [reply contributing/building-an-amp-extension.md]: Oh... Does adding `@const {string}` to the `EXPERIMENT` constant fix it?

Comment (caroqliu) [reply contributing/building-an-amp-extension.md]: No, I tried that also. I wonder why this check is enforced this way...

Comment (gmajoulet) [reply contributing/building-an-amp-extension.md]: Thanks for checking :))

**Pull request merged**